source/common/tracing/http_tracer_impl.cc

```diff
@@ -163,13 +163,36 @@ void HttpTracerUtility::finalizeSpan(Span& span, const Http::HeaderMap* request_
       span.setTag(Tracing::Tags::get().GuidXClientTraceId,
                   std::string(request_headers->ClientTraceId()->value().getStringView()));
     }
-
-    // Build tags based on the custom headers.
-    for (const Http::LowerCaseString& header : tracing_config.requestHeadersForTags()) {
-      const Http::HeaderEntry* entry = request_headers->get(header);
-      if (entry) {
-        span.setTag(header.get(), entry->value().getStringView());
+  }
+  // Build custom tags.
+  for (const CustomTag& custom_tag : tracing_config.customTags()) {
+    bool found = false;
```

Comment:
i think this might be slightly cleaner if you call into a helper that sets the tag and returns whether it was set, e.g.

```
bool hasValue = evaluateAndSetValue(custom_tag);
if (!hasValue && !custom_tag.default_value.empty()) {
  span.setTag(custom_tag.tag, custom_tag.default_value);
}
```

Reply:
fixed.